### PR TITLE
Combine `requiredPremisesCharacteristics` and `requiredRoomCharacteristics` on Bed Search into single `requiredCharacteristics`

### PR DESCRIPTION
**Pull request by brownbearsoftware**

### CODE REVIEW

**Comment by brownbearsoftware** [comment]
[Combine characteristics on Bed Search endpoint](https://trello.com/c/7FZ17VTX/1415-combine-characteristics-on-bed-search-endpoint)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BedSearchService.kt [hunk 1]
```diff
@@ -26,55 +26,31 @@ class BedSearchService(
...
-          "$.requiredPremisesCharacteristics" hasValidationError premisesCharacteristicErrors.joinToString(", ")
-        }
+        val characteristicErrors = mutableListOf<String>()
+        val premisesCharacteristicIds = mutableListOf<UUID>()
+        val roomCharacteristicIds = mutableListOf<UUID>()
 
-        val roomCharacteristicErrors = mutableListOf<String>()
-        val roomCharacteristics = requiredRoomCharacteristics.mapNotNull {
+        requiredCharacteristics.forEach {
           val characteristic = characteristicService.getCharacteristicByPropertyName(it)
```
Comment (pezholio) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BedSearchService.kt:42]: Could we fetch all these in one query?

Comment (pezholio) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BedSearchService.kt:42]: Something like (excuse the sketchy Kotlin)

```kotlin
val characteristics = characteristicService.getCharacteristicByPropertyNames(requiredCharacteristics)

requiredCharacteristics.forEach {
  val propertyName = it
  val characteristic = characteristics.find { it.property_name == propertyName}

  when {
    characteristic == null -> characteristicErrors += "$it doesNotExist"
    characteristic.matches(ServiceName.approvedPremises.value, "premises") -> premisesCharacteristicIds += characteristic.id
    characteristic.matches(ServiceName.approvedPremises.value, "room") -> roomCharacteristicIds += characteristic.id
    else -> characteristicErrors += "$it scopeInvalid"
  }
}
```

**Review by pezholio — COMMENTED** [review]
Just a quick comment re: performance

**Review by pezholio — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**